test/irmin-pack/concurrent.ml

```diff
@@ -0,0 +1,238 @@
+open Lwt.Infix
```

Comment:
I think this test can be considered separate from the layered store and split to a separate PR. I have some concerns as it stands.